src/main/java/io/mosip/mimoto/service/impl/CredentialServiceImpl.java

```diff
@@ -0,0 +1,253 @@
+        Map<String, CredentialDisplayResponseDto> credentialSubject = credentialsSupportedResponse.getCredentialDefinition().getCredentialSubject();
+        credentialSubject.keySet().forEach(VCProperty -> vcPropertiesFromWellKnown.put(VCProperty, credentialSubject.get(VCProperty).getDisplay().get(0).getName()));
+
+        Set<String> orderProperty = credentialsSupportedResponse.getOrder();
+
+        Set<String> fieldProperties = orderProperty == null ? vcPropertiesFromWellKnown.keySet() : orderProperty;
+        fieldProperties.forEach(vcProperty -> {
+            if(credentialProperties.get(vcProperty) != null) {
+                displayProperties.put(vcPropertiesFromWellKnown.get(vcProperty), credentialProperties.get(vcProperty));
+            }
+        });
+        return getPdfResourceFromVcProperties(displayProperties, credentialsSupportedResponse,  vcCredentialResponse,
+                issuerDTO.getDisplay().stream().map(d -> d.getLogo().getUrl()).findFirst().orElse(""));
+    }
+
+
+    private ByteArrayInputStream getPdfResourceFromVcProperties(LinkedHashMap<String, Object> displayProperties, CredentialsSupportedResponse credentialsSupportedResponse, VCCredentialResponse  vcCredentialResponse, String issuerLogoUrl) throws IOException, WriterException {
+        Map<String, Object> data = new HashMap<>();
+        LinkedHashMap<String, Object> rowProperties = new LinkedHashMap<>();
+        String backgroundColor = credentialsSupportedResponse.getDisplay().get(0).getBackgroundColor();
+        String textColor = credentialsSupportedResponse.getDisplay().get(0).getTextColor();
+        String credentialSupportedType = credentialsSupportedResponse.getDisplay().get(0).getName();
+        String face = vcCredentialResponse.getCredential().getCredentialSubject().get("face") != null ? (String) vcCredentialResponse.getCredential().getCredentialSubject().get("face") : null;
+
+        displayProperties.entrySet().stream()
+                .forEachOrdered(entry -> {
+                    if(entry.getValue() instanceof Map) {
+                        rowProperties.put(entry.getKey(), ((Map<?, ?>) entry.getValue()).get("value"));
+                    } else if(entry.getValue() instanceof List) {
+                        String value = "";
+                        if( ((List<?>) entry.getValue()).get(0) instanceof String) {
+                            value = ((List<String>) entry.getValue()).stream().reduce((field1, field2) -> field1 + ", " + field2 ).get();
+                        } else {
+                            value = (String) ((Map<?, ?>) ((List<?>) entry.getValue()).get(0)).get("value");
+                        }
+                        rowProperties.put(entry.getKey(), value);
+                    } else {
+                        rowProperties.put(entry.getKey(), entry.getValue());
+                    }
+                });
+
+        if(!credentialsSupportedResponse.getId().equals("MOSIPVerifiableCredential")) {
```

Comment:
Do we need to put check on credential type Id or issuer id?
Same will be applicable for mock mosip VC right? We won't get QR code for it

Reply:
We have removed the mock vc in collab, moreover mock can host all type of vc
Even we have host sunbird vc also from mock server, hence added it this way